insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/LocalInsightPlugin.java

```diff
@@ -1421,24 +1435,41 @@ private void processBolusDeliveredEvent(String serial, BolusDeliveredEvent event
                 event.getEventMinute(), event.getEventSecond(), event.getStartHour(), event.getStartMinute(), event.getStartSecond()) + timeOffset;
         InsightBolusID bolusID = insightDbHelper.getInsightBolusID(serial, event.getBolusID(), timestamp);
         if (bolusID == null || bolusID.getEndID() != null) {                        // TODO() Check if test EndID is necessary
+            if (bolusID == null) {
+                aapsLogger.debug(LTag.PUMP, "XXXX Delivered Bolus " + event.getBolusID() + "/" + event.getEventPosition() + " not recorded in Insight database");
+            } else {
+                aapsLogger.debug(LTag.PUMP, "XXXX Delivered Bolus " + bolusID.getId() + "/" + bolusID.getStartID() + "/" + bolusID.getEndID() + "=?" + event.getEventPosition() + " already recorded in Insight database");
+            }
             bolusID = new InsightBolusID(
                     startTimestamp,
                     serial,
                     event.getBolusID(),
                     bolusID == null ? event.getEventPosition() : bolusID.getStartID(),
                     event.getEventPosition());
+        } else {
+            aapsLogger.debug(LTag.PUMP, "XXXX Delivered Bolus " + bolusID.getId() + "!=" + event.getBolusID() + "/" + event.getEventPosition() + " already recorded in Insight database");
         }
         bolusID.setEndID(event.getEventPosition());
         insightDbHelper.createOrUpdate(bolusID);
         bolusID = insightDbHelper.getInsightBolusID(serial, event.getBolusID(), startTimestamp); // Line added to get id
         if (event.getBolusType() == BolusType.STANDARD || event.getBolusType() == BolusType.MULTIWAVE) {
-            pumpSync.syncBolusWithPumpId(
+            PumpSync.PumpState.Bolus lastRecordedBolus = pumpSync.expectedPumpState().getBolus();
+            aapsLogger.error(LTag.PUMP, "XXXX Last bolus in Database before update " + dateUtil.dateAndTimeString(lastRecordedBolus.getTimestamp()) + " amount: " + lastRecordedBolus.getAmount());
+            Boolean result = pumpSync.syncBolusWithPumpId(
```

Comment:
It's always here that I have the issue: 99% of the execution of this `pumpSync.syncBolusWithPumpId` are Ok, but about one or 2 times per day database is not updated and the last recorded bolus after execution is not correct.